ecommerce/api.py

```diff
@@ -458,12 +458,19 @@ def enroll_user_in_order_items(order):
         )
         edx_request_success = False
 
+    voucher = order.purchaser.vouchers.order_by("uploaded").last()
```

Comment:
Since product_id is now a field on `Voucher`, maybe this query should be modified to make sure the voucher is for the same product as what's in the order?  Something like:

```python
voucher = (
        order.purchaser.vouchers.filter(
            product_id__in=order.lines.all().values_list(
                "product_version__product__id", flat=True
            )
        )
        .order_by("uploaded")
        .last()
    )
```